### ISSUE TITLE
API requests failing

### ISSUE THREAD

**Issue by MrShakes** [DevExpress/testcafe-hammerhead#1838]
API requests have unfortunately been a bit flaky, requests are failing on Chrome and Firefox, work about two thirds of the time on Safari.
I'm clicking a button that makes requests to AWS cognito and AWS lambda and usually get a 400 or 500 error(works fine in Safari, most times). Did some research and tried creating a localhost https server and also telling chrome to allow insecure localhost as seen [here](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/connect-to-the-testcafe-server-over-https.html ) with no luck, bypass proxy also not working.

Testcafe version: 0.23.0
 Running tests in:
 - Chrome 70.0.3538 / Mac OS X 10.13.2
 - Firefox 62.0.0
 - Safari 11.0.2

**Comment by LavrovArtem** [comment]
I am afraid this information is not sufficient to find the cause of the issue. Could you please provide a page or a small example so that we can reproduce it on our side?

**Comment by MrShakes** [comment]
Ok, the code I'm using to try logging in with Chrome and Firefox:
```
fixture`Auth`.page`https://adhoc.greenlight.ai/`; 

test("Logging in and verifying it's the right user test 1", async t => {
    //Log in

    await t
        .typeText('#email-login', 'harry.potter@contractmagician.com')
        .typeText('#password-login', 'Passw0rd!')
        .click('#login-button');

    //Click right menu to navigate to profile
    await t
        .click('#zuser-menu')
        .click('#zuser-menu-profile');
});
```

Got 400 http error, and this one I used to try bypassing the error:
```
const createTestCafe = require('testcafe');
const selfSignedSertificate = require('openssl-self-signed-certificate');
let runner = null;

const sslOptions = {
    key: selfSignedSertificate.key,
    cert: selfSignedSertificate.cert
};

createTestCafe('localhost', 1337, 1338, sslOptions)
    .then(testcafe => {
        runner = testcafe.createRunner();
    })
    .then(() => {
        return (
            runner
                .src('/<path to test>/login.test.js')

                // Browsers restrict self-signed certificate usage unless you
                // explicitly set a flag specific to each browser.
                // For Chrome, this is '--allow-insecure-localhost'.
                .browsers('chrome --allow-insecure-localhost')
                .run()
        );
    });
```

And still didn't work, I'm getting a 500 http error in another part of my tests however if this is solved, could solve that too or lead to solving that. I've already provided system information above. It's worth noting that the tests always seem to freeze upon logging in.
Thanks.

**Comment by LavrovArtem** [comment]
Thanks for the details! I've reproduced this behavior with the `0.23.0` version. This issue is already fixed in the context of the #1828 and published in `testcafe@0.23.3-alpha.1`. And you don't necessarily need to use the `https` protocol to run your test.

**Comment by MrShakes** [comment]
Thanks, can confirm it's working now, still a bit flaky especially on Firefox, however can manage I guess.
